components/search_bar/search_bar.jsx

```diff
@@ -143,22 +143,22 @@ export default class SearchBar extends React.Component {
     }
 
     renderHintPopover(helpClass) {
-        if (!this.props.isCommentsPage && Utils.isMobile() && this.state.isPristine) {
-            return false;
+        if (!this.props.isCommentsPage && this.state.isPristine) {
```

Comment:
my point is, with this solution we are bringing back this https://mattermost.atlassian.net/browse/MM-5863 bug.

Actually investigating it a bit more `isCommentsPage` is not used anywhere, it will be always undefined. So now we are showing it only when isPristine. Before this commit we were showing it always for desktop and in mobile only when is not pristine. I'm not 100% sure if that bolean logic change is on purpose.

Actually, if the visibility is already handled by the css "visible" class added dynamically, you probably can render it always, without checking anything there. Maybe (I'm not sure if it is needed) add the `isPristine` to the visibility check.

I'm sorry for the rant. I think this component is a bit messy now. Needs some cleanup like remove unused properties and unneeded logic.

Reply:
Sounds cool. But when I checked this on mobile, the popover is not even being generated on mobile, it's not just about the visible class, the markup is not generated in that case.